backend/src/main/kotlin/org/loculus/backend/service/submission/SubmissionDatabaseService.kt

```diff
@@ -677,37 +690,97 @@ open class SubmissionDatabaseService(
+                )
+            }
+
+        val processingResultCounts = getProcessingResultCounts(groupIdsFilter, authenticatedUser, organism)
+
         return GetSequenceResponse(
-            sequenceEntries = pagedQuery
-                .map { row ->
-                    SequenceEntryStatus(
-                        accession = row[SequenceEntriesView.accessionColumn],
-                        version = row[SequenceEntriesView.versionColumn],
-                        status = Status.fromString(row[SequenceEntriesView.statusColumn]),
-                        groupId = row[SequenceEntriesView.groupIdColumn],
-                        submitter = row[SequenceEntriesView.submitterColumn],
-                        isRevocation = row[SequenceEntriesView.isRevocationColumn],
-                        submissionId = row[SequenceEntriesView.submissionIdColumn],
-                        dataUseTerms = DataUseTerms.fromParameters(
-                            DataUseTermsType.fromString(row[DataUseTermsTable.dataUseTermsTypeColumn]),
-                            row[DataUseTermsTable.restrictedUntilColumn],
-                        ),
-                    )
-                },
+            sequenceEntries = entries,
             statusCounts = statusCounts,
+            processingResultCounts = processingResultCounts,
+        )
+    }
+
+    /**
+     * How many processing results have errors, just warnings, or none?
+     * Considers only SequenceEntries that are PROCESSED.
+     */
+    private fun getProcessingResultCounts(
+        groupIdsFilter: List<Int>?,
+        authenticatedUser: AuthenticatedUser,
+        organism: Organism?,
+    ): Map<ProcessingResult, Int> {
+        val countBaseQuery = SequenceEntriesView
+            .select(
+                SequenceEntriesView.accessionColumn,
+                SequenceEntriesView.versionColumn,
+                SequenceEntriesView.errorsColumn,
+                SequenceEntriesView.warningsColumn,
+            )
+            .where { getGroupCondition(groupIdsFilter, authenticatedUser) }
+            .andWhere { SequenceEntriesView.statusIs(Status.PROCESSED) }
+
+        if (organism != null) {
+            countBaseQuery.andWhere { SequenceEntriesView.organismIs(organism) }
+        }
+
+        val errorCount = countBaseQuery.copy().andWhere { SequenceEntriesView.hasErrors }.count()
+        val warningCount = countBaseQuery.copy().andWhere {
+            SequenceEntriesView.hasWarnings and not(SequenceEntriesView.hasErrors)
+        }.count()
+        val noIssuesCount = countBaseQuery.copy().andWhere {
+            not(SequenceEntriesView.hasWarnings or SequenceEntriesView.hasErrors)
+        }.count()
+        val processingResultCounts = mapOf(
+            ProcessingResult.ERRORS to errorCount.toInt(),
+            ProcessingResult.WARNINGS to warningCount.toInt(),
+            ProcessingResult.NO_ISSUES to noIssuesCount.toInt(),
         )
```

Comment:
A group by query would be more efficient and require one instead of three queries. Maybe like this?

```kotlin
val processingResultType = case()
    .When(SequenceEntriesView.hasErrors, stringLiteral("HAS_ERRORS"))
    .When(SequenceEntriesView.hasWarnings, stringLiteral("HAS_WARNINGS"))
    .Else(stringLiteral("NO_ISSUES"))

val countQuery = SequenceEntriesView
    .select(processingResultType, Count(stringLiteral("*")))
    .where { getGroupCondition(groupIdsFilter, authenticatedUser) }
    .andWhere { SequenceEntriesView.statusIs(Status.PROCESSED) }

if (organism != null) {
    countQuery.andWhere { SequenceEntriesView.organismIs(organism) }
}

countQuery.groupBy(processingResultType)
…
```

Reply:
https://github.com/loculus-project/loculus/pull/3125/commits/942774ab09e7cf0810b82ad2cd236c47c4b2db58

Done!

Reply:
(Good suggestion by the way, thanks!)